opentelemetry-sdk/src/opentelemetry/sdk/metrics/export/aggregate.py

```diff
@@ -30,24 +31,43 @@ class Aggregator(abc.ABC):
     """
 
     def __init__(self, config=None):
-        self.current = None
-        self.checkpoint = None
-        if config:
+        self._lock = threading.Lock()
+        self.last_update_timestamp = 0
+        self.last_checkpoint_timestamp = 0
+        if config is not None:
             self.config = config
         else:
             self.config = {}
 
     @abc.abstractmethod
     def update(self, value):
         """Updates the current with the new value."""
+        self.last_update_timestamp = time_ns()
 
     @abc.abstractmethod
     def take_checkpoint(self):
         """Stores a snapshot of the current value."""
+        self.last_checkpoint_timestamp = time_ns()
 
     @abc.abstractmethod
     def merge(self, other):
         """Combines two aggregator values."""
+        self.last_update_timestamp = max(
+            self.last_update_timestamp, other.last_update_timestamp
+        )
+        self.last_checkpoint_timestamp = min(
```

Comment:
Should this be `max`?

Reply:
I was under the impression that this should be `min` in order to increase the time range since the "start" of the aggregator when it gets merged with another one. Please correct me if I am mistaken.

Reply:
Continued from the SIG:
It was my understanding that `last_checkpoint_timestamp` would work exactly like `last_update_timestamp` but for the `checkpoint` field. Upon merging (which I believe is there for when updates occur or when multiple updates occur at the same time), the "merged" record would be what we want the true record to contain. Conceptually, taking the latest checkpoint timestamp would be what we want, unless if I am misunderstanding what the purpose of `last_checkpoint_timestamp` is supposed to represent.

Reply:
Updated to `max`. :+1: